disperser/dataapi/metrics_handlers.go

```diff
@@ -142,3 +143,55 @@ func (s *server) getNonSigners(ctx context.Context, intervalSeconds int64) (*[]N
 
 	return &nonSignersObj, nil
 }
+
+func (s *server) getUnsignedBatches(ctx context.Context, intervalSeconds int64) (*UnsignedBatches, error) {
+	nonSigners, err := s.subgraphClient.QueryBatchNonSigningOperatorIdsInInterval(ctx, intervalSeconds)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		limit          = 1000
+		skip           = 0
+		totalOfBatches = 0
+	)
+
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop
+		default:
+			batches, err := s.subgraphClient.QueryBatchesWithLimit(ctx, limit, skip)
+			if err != nil {
+				s.logger.Error("Failed to query batches", "error", err)
+				return nil, err
+			}
+			totalOfBatches += len(batches)
+			if len(batches) == 0 {
+				break
+			}
+			skip += limit
+		}
+	}
+
+	percentage := make(map[string]float64, 0)
+	for operatorId, totalUnsignedBatches := range nonSigners {
+		if totalOfBatches < 1 {
```

Comment:
If `totalOfBatches` is 0, we can return immediately, right?

Reply:
Actually, we need to return `map[string]float64` and  `nonSigners` is type of `map[string]int`. 
If we return right away we don't have the operators' id inside map

Reply:
If `totalBatches` is 0, then all operators will have 0% nonsigning batches, right?

Also for the `map[string]float64`, we only need to return operators who have >0% nonsigning percentage.